### DISCUSSION TITLE
loops using for ... in and map

### DISCUSSION THREAD

**Discussion by coenttb** [elementary-swift/elementary#17] [Q&A] [answered]
Dear Sliemeobn,

```swift
extension List: HTML {
    public var content: some HTML {
        ul {
            for item in items {
                item
            }
        }
    }
}
```

This compiles and the syntax already looks quite nice. However, in some circumstances I prefer another way of writing this:

```swift
extension List: HTML {
    public var content: some HTML {
        ul {
            items.map { $0 }
        }
    }
}
```

Can you make this syntax work? I'm not an expert on resultBuilders, but I think this requires another build-block?

Thank you in advance for your time.

Best, Coen

**Comment by sliemeobn** [comment]
Hi @coenttb, thanks for getting in touch!

I had planned to add a `ForEach` element that can take sequences of HTML to better support all sorts of collections.
Supporting a map expression directly is an interesting idea, I don't think I ever came across a result builder that does that but it might actually work. Ideally it can "just" detect an expression that is a `Sequence<some HTML>`and make it work somehow.

**Reply by coenttb** [reply]
Would be amazing to see it work if it can "just" detect an expression that is a Sequence<some HTML>and make it work somehow.

As a note, I do see SwiftUI reach for the Group structure and Plot has .group(...) static function available. That would be a workable option for me, but your suggestion sounds much more elegant.

**Comment by sliemeobn** [comment, marked as answer]
hi @coenttb,
PR #26
wdyt?

**Reply by coenttb** [reply]
This looks like a great addition! You might want to check out the `HTMLForEach` feature in [my fork of swift-html from Point-Free](https://github.com/coenttb/swift-html), which I’m actively developing. The `PointFreeHtml` target was created by Point-Free, and you might find some inspiration there. I’ve reached out to them, and while they don’t plan on releasing or maintaining their version of swift-html as open-source yet, they may release it in the future.

Additionally, I’ve just made **[swift-css](https://github.com/coenttb/swift-css)** public. I'd love to hear if you find it useful or interesting, also with respect to Elementary!

**Reply by sliemeobn** [reply]
I have no desire to manually deal with css to be honest ; )
but I can see how that could be a cool combination
